sips/sip-02X/sip-02X-wallet-interface.md

```diff
@@ -0,0 +1,472 @@
+The Connect library should be go live with the new standard, once most major wallets have adopted the new standard.
+
+# Implementation
+
+## Notes on Serialization
+
+The WebBTC request standard is inspired by JSON-RPC 2.0 for serialization.
+To adhere to a generic serializability, the following notes are given.
+Enums are serialized as humand-readable strings.
+BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+Bytes are serialized as hex-encoded strings (without a 0x prefix).
+Predefined formats from previous SIPs are used where applicable.
+Addresses are serialized as Stacks c32-encoded strings.
+Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+## Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered recommendations for the wallet.
+The user/wallet may choose to ignore/override them.
+Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in settings like WebBTC (see WBIP-002) and other more Ethereum inspired domains.
+In other cases (e.g. WalletConnect), the namespace may already be given by meta-data (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+#### Common definitions
+
+The following definitions can be used in multiple methods (mainly for transfer and transaction methods).
+
+`params`
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'mocknet'`
+- `fee?`: `number | string` (anything parseable by the BigInt constructor)
+- `nonce?`: `number | string` (anything parseable by the BigInt constructor)
+- `attachment?`: `string` hex-encoded
+- `anchorMode?`: `'on-chain' | 'off-chain' | 'any'`
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+- `postConditionMode?`: `'allow' | 'deny'`
+- `sponsored?`: `boolean`, defaults to `false`
+- ~~`appDetails`~~ _removed_
+- ~~`onFinish`~~ _removed_
+- ~~`onCancel`~~ _removed_
+
+`where`
+
+- `PostCondition`: `string` hex-encoded
+
+---
+
+### Method `stx_transferStx`
+
+`params`
+
+- `recipient`: `string` address, Stacks c32-encoded
+- `amount`: `number | string` (anything parseable by the BigInt constructor)
+- `memo?`: `string`, defaults to `''`
+
+`result`
+
+- `txid`: `string` hex-encoded
+- `transaction`: `string` hex-encoded
+
+### Method `stx_transferFt`
+
+`todo: haven't existed yet, should we add them?`
+
+### Method `stx_transferNft`
+
+`todo: haven't existed yet, should we add them?`
+
+### Method `stx_callContract`
+
+`params`
+
+- `contract`: `string.string` address with contract name suffix, Stacks c32-encoded
+- `function`: `string`
```

Comment:
Is `arguments` reserved? I had this feeling as well, thought it might not be great, but wanted something shorter ☹️

Fine with going back to functionName, functionArgs. But happy to consider alternatives

Reply:
Changed to `functionName` and `functionArgs` again.
We could do something with `public` (since only public functions can be called. Any ideas welcom